src/es/templescanesp/src/eu/kanade/tachiyomi/extension/es/templescanesp/TempleScanEsp.kt

```diff
@@ -23,30 +25,67 @@ class TempleScanEsp :
         "Temple Scan",
         "https://aedexnox.vxviral.xyz",
         "es",
-        dateFormat = SimpleDateFormat("MMMM dd, yyyy", Locale("es")),
+        dateFormat = SimpleDateFormat("MMMM dd, yyyy", Locale("es"))
     ),
     ConfigurableSource {
 
     override val baseUrl get() = preferences.prefBaseUrl
+    private val preferences = getPreferences {
+        if (getString(DEFAULT_BASE_URL_PREF, null) == null) {
+            val currentUserUrl = getString(BASE_URL_PREF, super.baseUrl) ?: super.baseUrl
+            edit()
+                .putString(DEFAULT_BASE_URL_PREF, currentUserUrl)
+                .putBoolean(FETCH_DOMAIN_PREF, true)
+                .apply()
+        }
```

Comment:
The intention of this method is to override the user preference when the extension is updated. Also, the value of `getString(DEFAULT_BASE_URL_PREF, null)` technically can’t be null.


```suggestion
        this.getString(DEFAULT_BASE_URL_PREF, null).let { domain ->
            if (domain != super.baseUrl) {
                this.edit()
                    .putString(BASE_URL_PREF, super.baseUrl)
                    .putString(DEFAULT_BASE_URL_PREF, super.baseUrl)
                    .apply()
            }
        }
```